### PR TITLE
 Replace bootstrap-social with self-written CSS 

### PR DESCRIPTION
**Pull request by FestplattenSchnitzel**

### CODE REVIEW

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    color: #ffffff;
+
+    border: none;
+}
+.btn-social-icon>:first-child{
+    position: absolute;
+    left: 0;
+    top: 0;
+    bottom: 0;
+    width: 32px;
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:22]: Redundant, superseeded by the `width: 100%` below.
```suggestion
```

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    border-color: rgba(0, 0, 0, 0.2);
+}
+
+.btn-social-icon:active,.btn-social-icon.active, {
+    color: #fff;
+    background-color: color-mix(in hsl, black 16%, var(--btn-social-bg-color));
+    border-color: rgba(0, 0, 0, 0.2);
+}
+
+.btn-social-icon:active:hover,.btn-social-icon.active:hover,.btn-social-icon:active:focus,.btn-social-icon.active:focus,.btn-social-icon:active.focus,.btn-social-icon.active.focus {
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:50]: Why are all these combinations of pseudo and non-pseudo classes necessary?

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    color: #fff;
+    background-color: color-mix(in hsl, black, var(--btn-social-bg-color) 20%);
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:52]: What is the impact on non-recent browsers? how does it look when `color-mix` is not supported? As we found out it is a rather recent feature after all.

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    color: #fff;
+    background-color: color-mix(in hsl, black 16%, var(--btn-social-bg-color));
+    border-color: rgba(0, 0, 0, 0.2);
+}
+
+.btn-social-icon:active,.btn-social-icon.active, {
+    color: #fff;
+    background-color: color-mix(in hsl, black 16%, var(--btn-social-bg-color));
+    border-color: rgba(0, 0, 0, 0.2);
+}
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:48]: Can't we group these together? they are the exact same rules, are they not? :)

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    position: absolute;
+    left: 0;
+    top: 0;
+    bottom: 0;
+    width: 32px;
+    line-height: 34px;
+    font-size: 1.6em;
+    text-align: center;
+    border-right: 1px solid rgba(0, 0, 0, 0.2)
+    border: none;
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:27]: It's unclear what you want to achieve here.
First, it's missing a semicolon; so this rule is not applied due to the invalid value `1px solid rgba(0, 0, 0, 0.2) border: none`.
Second, when adding the semicolon, you would have the `border: none` overriding the border rule completely.

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    height: 34px;
+    width: 34px;
+    padding: 0;
+
+    background-color: var(--btn-social-bg-color);
+    color: #ffffff;
+
+    border: none;
+}
+.btn-social-icon>:first-child{
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:17]: First, I think `:first-child` is prematurely specific. The `.btn-social-icon` elements all only contain their respective icon `span` and nothing else.

Second, I think it should be much simpler to use flexbox for centering, let me have a stab at it and I'll get back to you.

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 7]
```diff
@@ -0,0 +1,67 @@
+.btn-social-icon {
+    position: relative;
+    padding-left: 44px;
+    text-align: left;
+    write-space: nowrap;
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:5]: typo, you meant `white-space`. However, you can most likely omit this, as the text the element contains is a single char.

File: sipa/static/css/agdsn-bootstrap-social.css [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    overflow: hidden;
+    text-overflow: ellipsis;
```
Comment (lukasjuhrich) [inline sipa/static/css/agdsn-bootstrap-social.css:7]: Same here. Is this necessary?

**Review by lukasjuhrich — COMMENTED** [review]
The result looks great, but I have the feeling that you have included way more CSS properties than necessary; most don't seem to be really doing anything.

File: sipa/static/css/agdsn-bootstrap-social.css (hunk 6 above)
Comment (lukasjuhrich) [reply sipa/static/css/agdsn-bootstrap-social.css:17]: Throw a `d-flex justify-content-center align-items-center` on all the `<div>`s containing the social links and then
```css
.btn-social-icon {
    height: 36px;
    width: 36px;
    padding: 0;
    background-color: var(--btn-social-bg-color);
    color: white;
    border: none;
}
.btn-social-icon > *{
    font-size: 1.6em;
}
.btn-social-icon > *::before {
    vertical-align: -1.5pt;
}
```
should suffice for the positioning.
The last thing is a slight optical correction (which bs-icons do themselves, but works to our disadvantage in this context).

File: sipa/static/css/agdsn-bootstrap-social.css (hunk 5 above)
Comment (FestplattenSchnitzel) [reply sipa/static/css/agdsn-bootstrap-social.css]: I think I just copy-pasted most of the CSS back then (should've noted that here though).

File: sipa/static/css/agdsn-bootstrap-social.css (hunk 6 above)
Comment (FestplattenSchnitzel) [reply sipa/static/css/agdsn-bootstrap-social.css]: Unfortunately, the icons still seem to be top aligned.  :(

File: sipa/static/css/agdsn-bootstrap-social.css (hunk 3 above)
Comment (FestplattenSchnitzel) [reply sipa/static/css/agdsn-bootstrap-social.css:23]: Here is the compatibility matrix [0], will try to test with some old enough browser.

[0] https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix#browser_compatibility

File: sipa/static/css/agdsn-bootstrap-social.css (hunk 6 above)
Comment (lukasjuhrich) [reply sipa/static/css/agdsn-bootstrap-social.css]: > Unfortunately, the icons still seem to be top aligned. :(



> Unfortunately, the icons still seem to be top aligned. :(

What do you mean by "still top aligned"? You never claimed they were top aligned before, and I never saw them top aligned.

Comment (lukasjuhrich) [reply sipa/static/css/agdsn-bootstrap-social.css]: Resolved offline (734e80c00f337fd6f4021690d83fbe0ed853fa96)

File: sipa/static/css/agdsn-bootstrap-social.css (hunk 3 above)
Comment (lukasjuhrich) [reply sipa/static/css/agdsn-bootstrap-social.css:23]: Ignoring, finding an older browser takes too much time – and: it is the hover rule, anyway.

**Pull request merged**